PP-1027: Add Man page functionality test in PTL smoke test

Mahalty opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Issue-ID
<!--- Replace XXXX below (2 places) with the actual JIRA id -->
* **[PP-1027](https://pbspro.atlassian.net/browse/PP-1027)**

#### Problem description
* Test case for "man page functionality" was missing from smoke test

#### Cause / Analysis
* Briefly describe root cause/analysis of the problem

#### Solution description
* Added "Man page functionality" test in PTL smoke test

#### Checklist:
<!--- Use the preview button to see the checkboxes/links properly. -->
<!--- Go over the following points, and put an `x` (without spaces around it) in the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.


__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PBSPro/pbspro?pullRequest=414) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/PBSPro/pbspro?pullRequest=414) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PBSPro/pbspro?pullRequest=414) it.</sub>

bhroam commented on test/tests/pbs_smoketest.py:
```diff
@@ -1458,3 +1458,20 @@ def test_interactive_job(self):
+        manpath = os.path.join(pbs_conf['PBS_EXEC'], "share", "man")
+        pbs_cmnds = ["pbsnodes", "qsub"]
+        for pbs_cmd in pbs_cmnds:
+            cmd = ["export MANPATH=$MANPATH:%s;" % manpath, "man %s" % pbs_cmd]
```
This can be a little more concise.  If you put VAR=VAL in front of a command, it will be added to the environment for that command only.  You can do this all in one command instead of two.  I think you'll still need as_script=True.  The other option is to directly add MANPATH do the environment via os.environ() before running the command.  This way you'll only do it once.

bhroam commented on test/tests/pbs_smoketest.py:
```diff
@@ -1458,3 +1458,20 @@ def test_interactive_job(self):
+        """
+        pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+        manpath = os.path.join(pbs_conf['PBS_EXEC'], "share", "man")
+        pbs_cmnds = ["pbsnodes", "qsub"]
```
Why only two commands?  It's a for loop, you might as well toss a couple more in.  Maybe a daemon and qmgr?

bhroam commented on test/tests/pbs_smoketest.py:
```diff
@@ -1458,3 +1458,20 @@ def test_interactive_job(self):
+            cmd = ["export MANPATH=$MANPATH:%s;" % manpath, "man %s" % pbs_cmd]
+            rc = self.du.run_cmd(cmd=cmd, as_script=True)
+            msg = "Error while retrieving man page of %s" % pbs_cmd
+            msg += "command:%s" % rc['err']
```
Could you add a space after the ':'?

anamikau commented:
Hey Bhroam, I would like to comment in the 2 commands part. This test is just to check that man is not throwing error for PBS commands. So one or 2 tests are fine. Plus it is a smoke test. If we want more then they should go in separate regressions for man.

Regards,
Anamika

Sent from my iPhone

On Oct 10, 2017, at 11:27 AM, Bhroam Mann <notifications@github.com<mailto:notifications@github.com>> wrote:


@bhroam requested changes on this pull request.

________________________________

In test/tests/pbs_smoketest.py<https://github.com/PBSPro/pbspro/pull/414#discussion_r143812564>:

> @@ -1458,3 +1458,20 @@ def test_interactive_job(self):
         self.server.expect(JOB, {'job_state': 'R'}, id=jid)
         self.server.delete(jid)
         self.server.expect(JOB, 'queue', op=UNSET, id=jid)
+
+    def test_man_pages(self):
+        """
+        Test basic functionality of man pages
+        """
+        pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+        manpath = os.path.join(pbs_conf['PBS_EXEC'], "share", "man")
+        pbs_cmnds = ["pbsnodes", "qsub"]
+        for pbs_cmd in pbs_cmnds:
+            cmd = ["export MANPATH=$MANPATH:%s;" % manpath, "man %s" % pbs_cmd]


This can be a little more concise. If you put VAR=VAL in front of a command, it will be added to the environment for that command only. You can do this all in one command instead of two. I think you'll still need as_script=True. The other option is to directly add MANPATH do the environment via os.environ() before running the command. This way you'll only do it once.

________________________________

In test/tests/pbs_smoketest.py<https://github.com/PBSPro/pbspro/pull/414#discussion_r143812852>:

> @@ -1458,3 +1458,20 @@ def test_interactive_job(self):
         self.server.expect(JOB, {'job_state': 'R'}, id=jid)
         self.server.delete(jid)
         self.server.expect(JOB, 'queue', op=UNSET, id=jid)
+
+    def test_man_pages(self):
+        """
+        Test basic functionality of man pages
+        """
+        pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+        manpath = os.path.join(pbs_conf['PBS_EXEC'], "share", "man")
+        pbs_cmnds = ["pbsnodes", "qsub"]


Why only two commands? It's a for loop, you might as well toss a couple more in. Maybe a daemon and qmgr?

________________________________

In test/tests/pbs_smoketest.py<https://github.com/PBSPro/pbspro/pull/414#discussion_r143813121>:

> @@ -1458,3 +1458,20 @@ def test_interactive_job(self):
         self.server.expect(JOB, {'job_state': 'R'}, id=jid)
         self.server.delete(jid)
         self.server.expect(JOB, 'queue', op=UNSET, id=jid)
+
+    def test_man_pages(self):
+        """
+        Test basic functionality of man pages
+        """
+        pbs_conf = self.du.parse_pbs_config(self.server.shortname)
+        manpath = os.path.join(pbs_conf['PBS_EXEC'], "share", "man")
+        pbs_cmnds = ["pbsnodes", "qsub"]
+        for pbs_cmd in pbs_cmnds:
+            cmd = ["export MANPATH=$MANPATH:%s;" % manpath, "man %s" % pbs_cmd]
+            rc = self.du.run_cmd(cmd=cmd, as_script=True)
+            msg = "Error while retrieving man page of %s" % pbs_cmd
+            msg += "command:%s" % rc['err']


Could you add a space after the ':'?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/PBSPro/pbspro/pull/414#pullrequestreview-68380136>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AQhxDECAtnN43VKbFCtoNw9rAoVumoeTks5sq7bLgaJpZM4Pz8TD>.

bhroam commented:
Travis is failing because the man command is not installed on the opensuse docker container.  We'll need to get travis updates before this PR can check in.

Mahalty commented:
@bhroam I have addressed your review comments and attached Latest execution-logs in the ticket.
Also I have filed [PP-1029](https://pbspro.atlassian.net/browse/PP-1029) to update Travis

bhroam commented on test/tests/pbs_smoketest.py:
```diff
@@ -1466,11 +1466,12 @@ def test_man_pages(self):
         pbs_conf = self.du.parse_pbs_config(self.server.shortname)
         manpath = os.path.join(pbs_conf['PBS_EXEC'], "share", "man")
         pbs_cmnds = ["pbsnodes", "qsub"]
+        os.environ['MANPATH'] = '%s' % manpath 
```
You don't need to do string formatting for just '%s'.  You can just directly set os.environ['MANPATH'] = manpath.

bhroam commented:
Codacy is complaining about trailing whitespace.  I believe pep8 will also complain.

bhroam commented:
I merged in the PR.
